jetty-core/jetty-util/src/test/java/org/eclipse/jetty/util/resource/PathResourceTest.java

```diff
@@ -413,6 +416,44 @@ public void testSymlink(WorkDir workDir) throws Exception
         }
     }
 
+    @Test
+    public void testBrokenSymlink(WorkDir workDir) throws Exception
+    {
+        Path testDir = workDir.getEmptyPathDir();
+        Path resourcePath = testDir.resolve("resource.txt");
+        IO.copy(MavenTestingUtils.getTestResourcePathFile("resource.txt").toFile(), resourcePath.toFile());
+        Path symlinkPath = Files.createSymbolicLink(testDir.resolve("symlink.txt"), resourcePath);
+
+        PathResource fileResource = new PathResource(resourcePath);
+        assertTrue(fileResource.exists());
+        PathResource symlinkResource = new PathResource(symlinkPath);
+        assertTrue(symlinkResource.exists());
+
+        // Their paths are not equal but not their canonical paths are.
+        assertThat(fileResource.getPath(), not(equalTo(symlinkResource.getPath())));
+        assertThat(fileResource.getPath(), equalTo(symlinkResource.getTargetPath()));
+        assertFalse(fileResource.isAlias());
+        assertTrue(symlinkResource.isAlias());
+        assertTrue(fileResource.exists());
+        assertTrue(symlinkResource.exists());
+
+        // After deleting file the Resources do not exist even though symlink file exists.
+        Files.delete(resourcePath);
```

Comment:
This delete needs protection for users on filesystems that don't immediately delete.
Test that the `resourcePath` was actually deleted in an assumption.